doc/introduction/program_capture_sharp_bits.rst

```diff
@@ -0,0 +1,384 @@
+cases that magically work, but one shouldn't extrapolate beyond those particular 
+cases.
+
+Instead, it is best practice to `use jax.vmap <https://docs.jax.dev/en/latest/_autosummary/jax.vmap.html>`__:
+
+.. code-block:: python
+
+    dev = qml.device("default.qubit", wires=1)
+
+    @qml.qnode(dev)
+    def circuit(x):
+        qml.RX(x, wires=0)
+        return qml.expval(qml.Z(0))
+
+>>> x = jnp.array([0.1, 0.2, 0.3])
+>>> vmap_circuit = jax.vmap(circuit)
+>>> vmap_circuit(x)
+Array([0.9950042 , 0.9800666 , 0.95533645], dtype=float32)
+
+More information for using ``jax.vmap`` can be found in the 
+`JAX documentation <https://docs.jax.dev/en/latest/_autosummary/jax.vmap.html#jax.vmap>`__.
+
+Transforms
+----------
+
+One of the core features of PennyLane is modularity, which has allowed users to 
+transform QNodes in a NumPy-like way and to create their own transforms with ease. 
+Your favourite transforms will still work with program-capture enabled (including
+custom transforms), but **decorating QNodes with just ``@transform_name`` will not 
+work** and will give a vague error. Additionally, decorating QNodes with the experimental 
+:func:`~.pennylane.capture.expand_plxpr_transforms` decorator is required.
+
+Consider the following toy example, which shows a tape-based transform that shifts 
+all :class:`~.pennylane.RX`` gates to the end of a circuit.
+
+.. code-block:: python
+
+    @qml.transform
+    def shift_rx_to_end(tape):
+        """Transform that moves all RX gates to the end of the operations list."""
+        new_ops, rxs = [], []
+
+        for op in tape.operations:
+            if isinstance(op, qml.RX):
+                rxs.append(op)
+            else:
+                new_ops.append(op)
+        
+        operations = new_ops + rxs
+        new_tape = tape.copy(operations=operations)
+        return [new_tape], lambda res: res[0]
+
+Decorating with just ``@shift_rx_to_end`` will not work, and will give a vague error:
+
+.. code-block:: python
+
+    @shift_rx_to_end
+    @qml.qnode(qml.device("default.qubit", wires=1))
+    def circuit():
+        qml.RX(0.1, wires=0)
+        qml.H(wires=0)
+        return qml.state()
+
+>>> print(qml.draw(circuit)())
+...
+NotImplementedError: 
+
+A requirement for tape transforms to be compatible with program capture is to further 
+decorate QNodes with the experimental :func:`~.pennylane.capture.expand_plxpr_transforms` 
+decorator:
+
+.. code-block:: python
+
+    @qml.capture.expand_plxpr_transforms
+    @shift_rx_to_end
+    @qml.qnode(qml.device("default.qubit", wires=1))
+    def circuit():
+        qml.RX(0.1, wires=0)
+        qml.H(wires=0)
+        return qml.state()
+
+>>> print(qml.draw(circuit)())
+0: ──H──RX(0.10)─┤  State
+
+Dynamic variables and transforms
+================================
+
+Some transforms in the :module:`~.pennylane.transforms` API have bespoke plxpr implementations
+that *directly* transform jaxpr:
```

Comment:
This is a bit confusing (leans into technical details) -- better to state the main take away for users. E.g.,

```suggestion
Some transforms in the :doc:`/code/qml_transforms` module have natively support program capture mode:
```